src/core/annotation.js

```diff
@@ -3880,6 +3902,22 @@ class PopupAnnotation extends Annotation {
 
     this.data.open = !!dict.get("Open");
   }
+
+  static createNewDict(annotation, xref, _params) {
+    const { oldAnnotation, rect, parent } = annotation;
+    const popup = oldAnnotation || new Dict(xref);
+    popup.setIfNotExists("Type", Name.get("Annot"));
+    popup.setIfNotExists("Subtype", Name.get("Popup"));
+    popup.setIfNotExists("Open", false);
+    popup.setIfArray("Rect", rect);
+    popup.setIfDefined("Parent", parent);
```

Comment:
AFAIK popup annotations can't live on their own without a parent because nothing could trigger them then, so could we perhaps also use `popup.set("Parent", parent")` here since the parent should always be defined, or are there cases possible where there is no parent?